include/deal.II/rol/vector_adaptor.h

```diff
@@ -0,0 +1,478 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2017 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal.II distribution.
+//
+//---------------------------------------------------------------
+
+#ifndef dealii_rol_vector_adaptor_h
+#define dealii_rol_vector_adaptor_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_TRILINOS
+#include <ROL_Vector.hpp>
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/lac/vector.h>
+
+#include <type_traits>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+
+using namespace dealii;
+
+
+/**
+ * A namespace that provides an interface to the
+ * <a href="https://trilinos.org/docs/dev/packages/rol/doc/html/index.html">
+ * Rapid Optimization Library</a> (ROL).
+ */
+namespace rol
```

Comment:
We tend to capitalise the first letter of a namespace, but since its an abbreviation itself this should be `namespace ROL`.

Reply:
I initially intended to use `namespace ROL`, but it conflicts with the ROL library's native `namespace ROL` and could lead to confusion. May be I could use `namespace Rol`?

Reply:
Thats unfortunate. Yours is really `dealii::ROL` but then I suppose you might have to make this qualification everywhere. In that case I'd be content with `Rol`.